pennylane/clusters.py

```diff
@@ -0,0 +1,476 @@
+              through the QNodeCluster.The QNodeCluster accepts and returns Torch tensors.
+
+            * ``interface='tf'``: Allows TensorFlow in eager mode to backpropogate
+              through the QNodeCluster. The QNodeCluster accepts and returns
+              TensorFlow ``tf.Variable`` and ``tf.tensor`` objects.
+
+            * ``None``: The QNodeCluster accepts default Python types
+              (floats, ints, lists) as well as NumPy array arguments,
+              and returns NumPy arrays. It does not connect to any
+              machine learning library automatically for backpropagation.
+
+        diff_method (str, None): the method of differentiation to use in the created QNodeCluster.
+
+            * ``"best"``: Best available method. Uses the device directly to compute
+              the gradient if supported, otherwise will use the analytic parameter-shift
+              rule where possible with finite-difference as a fallback.
+
+            * ``"parameter-shift"``: Use the analytic parameter-shift
+              rule where possible with finite-difference as a fallback.
+
+            * ``"finite-diff"``: Uses numerical finite-differences.
+
+            * ``None``: a non-differentiable QNodeCluster is returned.
+
+    Returns:
+        QNodeCluster: a cluster of QNodes executing the circuit template with
+        the specified measurements
+
+    **Example:**
+
+    Let's define a custom two-wire template:
+
+    .. code-block:: python
+
+        def my_template(params, wires, **kwargs):
+            for i in range(2):
+                qml.RX(params[i], wires=wires[i])
+
+            qml.CNOT(wires=wires)
+
+    We want to compute the expectation values over the following list
+    of two-qubit observables:
+
+    >>> obs_list = [qml.PauliX(0) @ qml.PauliZ(1), qml.PauliZ(0) @ qml.PauliX(1)]
+
+    This can be easily done with the ``map`` function:
+
+    >>> dev = qml.device("default.qubit", wires=2)
+    >>> qnodes = qml.map(my_template, obs_list, dev, measure="expval")
+    >>> qnodes([0.54, 0.12])
+    array([-0.06154835  0.99280864])
+    """
+    if not callable(template):
+        raise ValueError("Could not create QNodes. The template is not a callable function.")
+
+    qnodes = QNodeCluster()
+
+    if not isinstance(device, Sequence):
+        # broadcast the single device over all observables
+        device = [device] * len(observables)
+
+    if not isinstance(measure, (list, tuple)):
+        # broadcast the single measurement over all observables
+        measure = [measure] * len(observables)
+
+    for obs, m, dev in zip(observables, measure, device):
+        # Generate QNodes from all pairs of observables, measurements, and devices.
+        if not isinstance(obs, Observable):
+            raise ValueError("Could not create QNodes. Some or all observables are not valid.")
+
+        wires = list(range(dev.num_wires))
+
+        def circuit(
+            params, _obs=obs, _m=m, _wires=wires, **kwargs
```

Comment:
If ``_wires=wires`` is removed from the signature and ``_wires = None`` is defined on line 135 (adding newlines before and after), the pylint directives can be removed (actually the second one can be removed right now just as it is)

Reply:
This is the main reason behind why that solution doesn't seem to work: https://docs.python-guide.org/writing/gotchas/#late-binding-closures

Reply:
You are right, thanks so much for this! :)